src/holofuel/pages/Profile/Profile.js

```diff
@@ -20,10 +20,16 @@ function Card ({ title, subtitle, children }) {
 }
 
 export default function Profile () {
-  const { loading, data: { holofuelUser: { id, nickname } = {} } = {} } = useQuery(HolofuelUserQuery, { fetchPolicy: 'cache-and-network' })
+  const { loading, data: { holofuelUser: { id, nickname } = {} } = {}, refetch: refetchHolofuelUser } = useQuery(HolofuelUserQuery, { fetchPolicy: 'cache-and-network' })
   const [updateUser] = useMutation(HolofuelUpdateUserMutation)
   const [optimisitcNickname, setOptimisitcNickname] = useState('')
 
+  useEffect(() => {
+    if (optimisitcNickname && !nickname) {
+      refetchHolofuelUser()
+    }
+  }, [optimisitcNickname, nickname, refetchHolofuelUser])
```

Comment:
this might cause an infinite loop if for some reason the nickname didn't make it into the backend. Let's discuss an approach.